uvtools/plot.py

```diff
@@ -466,3 +469,265 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+            "Please ensure the plot type chosen is supported. The supported " \
+            "types are : {types}".format(types=list(plot_types.keys()))
+
+    # now make a dictionary of the transformed visibilities
+    visibilities = {plot : {label : xform(vis)
+                            for label, vis in zip(("vis1", "vis2"), (vis1, vis2))}
+                            for plot, xform in plot_types.items()
+                            if plot in plot_type} # but only use desired transforms
+
+
+    # import matplotlib, setup the figure
+    import matplotlib.pyplot as plt
+    figsize = (4 * 3, 3 * len(plot_type)) # (4,3) figsize for each plot
+    fig = plt.figure(figsize=figsize)
+    axes = fig.subplots(len(plot_type), 3)
+    axes = [axes,] if len(axes.shape) == 1 else axes # avoid bug for single row
+    axes[0][0].set_title("Amplitude Difference", fontsize=12)
+    axes[0][1].set_title("Phase Difference", fontsize=12)
+    axes[0][2].set_title("Amplitude of Complex Difference", fontsize=12)
+
+    # helper function for getting the extent of axes
+    extent = lambda xvals, yvals : (xvals[0], xvals[-1], yvals[-1], yvals[0])
+
+    # loop over items in visibilities and plot them
+    for i, item in enumerate(visibilities.items()):
+        # extract visibilities, get diffs
+        visA, visB = item[1].values()
+        diffs = (utils.diff(visA, visB, 'abs'), 
+                 utils.diff(visA, visB, 'phs'),
+                 utils.diff(visA, visB, 'complex'))
+
+        # extract parameters
+        ykey, xkey = item[0].split("_vs_") # keys for choosing parameters
+        xvals, yvals = plot_params[xkey], plot_params[ykey]
+
+        # get labels
+        xlabel, ylabel = labels[xkey], labels[ykey]
+
+        # plot stuff
+        for ax, diff in zip(axes[i], diffs):
+            # set labels
+            ax.set_xlabel(xlabel, fontsize=12)
+            ax.set_ylabel(ylabel, fontsize=12)
+
+            # plot waterfall and add a colorbar
+            fig.sca(ax)
+            cax = waterfall(diff, mode="real", cmap='viridis', 
+                            extent=extent(xvals, yvals))
+            fig.colorbar(cax)
+
+    # display the figure
+    plt.tight_layout()
+    plt.show()
+
+    # save if desired
+    if save_path is not None:
+        utils.savefig(fig, save_path)
+
+def plot_diff_uv(uvd1, uvd2, pol=None, speedup=True,
+                 skip_check=False, save_path=None,
+                 resolution=50):
+    """Summary plot for difference between visibilities.
+
+    Parameters
+    ----------
+    uvd1, uvd2 : pyuvdata.UVData
+        Input UVData objects which contain the visibilities to be differenced 
+        and any relevant metadata.
+
+    pol : str, None, optional
+        String specifying which polarization to be used. Must be one of the
+        polarizations listed in the UVData.get_pols() method for both
+        `uvd1` and `uvd2`. Default is to use the 0th polarization.
+
+    speedup : bool, optional
```

Comment:
Might be worth explaining what the fast method is doing that makes it fast, so users can understand if this might cause issues in how they interpret the plots.

Reply:
Ah, this used to actually do something (when I had two implementations for plotting), but now there's only one way of generating the arrays to plot. The new method is also significantly faster than what the faster method of the two old implementations were. (It now uses `scipy.stats.binned_statistic_2d`. The previous implementation wasn't exactly doing the right thing.)